torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -2913,6 +2918,7 @@ def _test_scatter_helper(
             self._barrier()
 
         @sandcastle_skip_if(BACKEND == "nccl", "Nccl does not support CPU tensors")
+        @sandcastle_skip_if(BACKEND == "ucc", "CPU tensor ops not supported by UCP TL")
```

Comment:
Curious: what would happen if a user indeed makes a scatter call with CPU tensors?

Reply:
User will get an error, for example:
```
RuntimeError: [/var/lib/jenkins/workspace/torch/csrc/distributed/c10d/ProcessGroupUCC.cpp:463] [Rank 1][ProcessGroupUCC-1][READY]failed to init collective, error code -1: Operation is not supported, system error code 11
Exception raised from enqueue_collective at /var/lib/jenkins/workspace/torch/csrc/distributed/c10d/ProcessGroupUCC.cpp:463 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6b (0x7ff89899a60b in /opt/conda/lib/python3.10/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xce (0x7ff898995fde in /opt/conda/lib/python3.10/site-packages/torch/lib/libc10.so)
frame #2: <unknown function> + 0x1d9148 (0x7ff8ae10c148 in /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_cuda_cpp.so)
frame #3: c10d::ProcessGroupUCC::scatter(std::vector<at::Tensor, std::allocator<at::Tensor> >&, std::vector<std::vector<at::Tensor, std::allocator<at::Tensor> >, std::allocator<std::vector<at::Tensor, std::allocator<at::Tensor> > > >&, c10d::ScatterOptions const&) + 0x77b (0x7ff8ae11f5ab in /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_cuda_cpp.so)
frame #4: <unknown function> + 0x4548189 (0x7ff8a6647189 in /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #5: <unknown function> + 0x454da04 (0x7ff8a664ca04 in /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #6: <unknown function> + 0x4559348 (0x7ff8a6658348 in /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #7: c10d::ops::scatter(c10::intrusive_ptr<c10d::ProcessGroup, c10::detail::intrusive_target_default_null_type<c10d::ProcessGroup> > const&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, std::vector<std::vector<at::Tensor, std::allocator<at::Tensor> >, std::allocator<std::vector<at::Tensor, std::allocator<at::Tensor> > > > const&, c10d::ScatterOptions const&) + 0x2ac (0x7ff8a664b51c in /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #8: <unknown function> + 0x9b21cd (0x7ff8b048e1cd in /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_python.so)
frame #9: <unknown function> + 0x242487 (0x7ff8afd1e487 in /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_python.so)
frame #10: <unknown function> + 0x17ffa4 (0x55cc596f2fa4 in /opt/conda/bin/python)
frame #11: _PyObject_MakeTpCall + 0x15e (0x55cc596b552e in /opt/conda/bin/python)
frame #12: <unknown function> + 0xaea74 (0x55cc59621a74 in /opt/conda/bin/python)
frame #13: <unknown function> + 0x1052a5 (0x55cc596782a5 in /opt/conda/bin/python)
frame #14: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #15: _PyObject_Call + 0xf7 (0x55cc596bae97 in /opt/conda/bin/python)
frame #16: _PyEval_EvalFrameDefault + 0x26e4 (0x55cc59756774 in /opt/conda/bin/python)
frame #17: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #18: <unknown function> + 0x104425 (0x55cc59677425 in /opt/conda/bin/python)
frame #19: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #20: <unknown function> + 0x104425 (0x55cc59677425 in /opt/conda/bin/python)
frame #21: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #22: <unknown function> + 0x18fc9b (0x55cc59702c9b in /opt/conda/bin/python)
frame #23: <unknown function> + 0x1050b5 (0x55cc596780b5 in /opt/conda/bin/python)
frame #24: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #25: <unknown function> + 0x1049f3 (0x55cc596779f3 in /opt/conda/bin/python)
frame #26: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #27: <unknown function> + 0x18fc9b (0x55cc59702c9b in /opt/conda/bin/python)
frame #28: <unknown function> + 0x104425 (0x55cc59677425 in /opt/conda/bin/python)
frame #29: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #30: <unknown function> + 0x18fc9b (0x55cc59702c9b in /opt/conda/bin/python)
frame #31: <unknown function> + 0x104425 (0x55cc59677425 in /opt/conda/bin/python)
frame #32: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #33: <unknown function> + 0x1049f3 (0x55cc596779f3 in /opt/conda/bin/python)
frame #34: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #35: <unknown function> + 0x18fd4e (0x55cc59702d4e in /opt/conda/bin/python)
frame #36: _PyObject_Call + 0x20a (0x55cc596bafaa in /opt/conda/bin/python)
frame #37: _PyEval_EvalFrameDefault + 0x26e4 (0x55cc59756774 in /opt/conda/bin/python)
frame #38: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #39: <unknown function> + 0x1049f3 (0x55cc596779f3 in /opt/conda/bin/python)
frame #40: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #41: <unknown function> + 0x1049f3 (0x55cc596779f3 in /opt/conda/bin/python)
frame #42: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #43: <unknown function> + 0x104425 (0x55cc59677425 in /opt/conda/bin/python)
frame #44: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #45: <unknown function> + 0x1050b5 (0x55cc596780b5 in /opt/conda/bin/python)
frame #46: <unknown function> + 0x18f742 (0x55cc59702742 in /opt/conda/bin/python)
frame #47: PyEval_EvalCode + 0x88 (0x55cc59711578 in /opt/conda/bin/python)
frame #48: <unknown function> + 0x2548fb (0x55cc597c78fb in /opt/conda/bin/python)
frame #49: <unknown function> + 0x28d3a5 (0x55cc598003a5 in /opt/conda/bin/python)
frame #50: PyRun_StringFlags + 0x7a (0x55cc598051ca in /opt/conda/bin/python)
frame #51: PyRun_SimpleStringFlags + 0x3c (0x55cc5980522c in /opt/conda/bin/python)
frame #52: Py_RunMain + 0x46d (0x55cc598056cd in /opt/conda/bin/python)
frame #53: Py_BytesMain + 0x39 (0x55cc598057d9 in /opt/conda/bin/python)
frame #54: __libc_start_main + 0xe7 (0x7ff8e7d9ec87 in /lib/x86_64-linux-gnu/libc.so.6)
frame #55: <unknown function> + 0x2125d4 (0x55cc597855d4 in /opt/conda/bin/python)
```